src/main/java/frc/robot/subsystems/ControlPanelSubsystem.java

```diff
@@ -0,0 +1,79 @@
+package frc.robot.subsystems;
+
+import java.util.List;
+
+import com.revrobotics.ColorSensorV3;
+import com.revrobotics.ColorMatch;
+
+import com.ctre.phoenix.motorcontrol.can.VictorSPX;
+
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.I2C.Port;
+import edu.wpi.first.wpilibj.util.Color;
+
+public class ControlPanelSubsystem extends SubsystemBase {
+
+    public VictorSPX cpMotor;
+    public DoubleSolenoid cpSolenoid;
+    public ColorSensorV3 colorSensor;
+    public DriverStation ds;
+    public ColorMatch colorMatcher;
+
+    Color turnToColor = null;
+    public Color detectedColor = null;
+    public Color fmsColor = null;
+    Color lastColor = null;
+
+    List<String> colorNames;
+    public List<Color> colors;
+
+    //TODO: Both of these were "will_reset_to" variables in python, but I couldn't find a java alternative
+    public int speed = 0;
+    public DoubleSolenoid.Value solenoidState = DoubleSolenoid.Value.kReverse;
+
+    public ControlPanelSubsystem(){
+        cpMotor = new VictorSPX(2);
+        cpSolenoid = new DoubleSolenoid(5, 4);
+        colorSensor = new ColorSensorV3(Port.kOnboard);
+        ds = DriverStation.getInstance();
+        colorMatcher = new ColorMatch();
+
+        colorNames.add("B");
+        colorNames.add("G");
+        colorNames.add("R");
+        colorNames.add("Y");
+
+        colors.add(new Color(0.175, 0.469, 0.356));
+        colors.add(new Color(0.188, 0.503, 0.310));
+        colors.add(new Color(0.325, 0.453, 0.221));
+        colors.add(new Color(0.269, 0.550, 0.181));
```

Comment:
Use a HashMap for these, don't use two different lists.

Reply:
@ArchdukeTim The first time I was writing this, I looked into HashMaps, but I wasn't able to find an ordered HashMap. If you know a way to do this, please let me know.

Reply:
LinkedHashMap 😉

Reply:
I looked at LinkedHashMaps, but I couldn't find a way to get a key/object at a certain index, or find the index of a certain key/object.